### ISSUE TITLE
SNOW-451953: Data upload not working after upgrading from 2.5.1 to 2.6.0

### ISSUE THREAD

**Issue by jonashaag** [snowflakedb/snowflake-connector-python#854]
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?
    Python 3.8.10 | packaged by conda-forge | (default, May 11 2021, 07:01:05) [GCC 9.3.0]
2. What operating system and processor architecture are you using?
   Linux-5.4.0-1055-aws-x86_64-with-glibc2.10
3. What are the component versions in the environment (`pip freeze`)?
   ```
   $ mamba list | grep -E 'snowfl|boto|arrow'                                                                                                                                             
   arrow                     1.1.1              pyhd8ed1ab_0    conda-forge
   arrow-cpp                 3.0.0           py38h4026a5f_22_cpu    conda-forge
   boto3                     1.18.28            pyhd8ed1ab_0    conda-forge
   botocore                  1.21.28            pyhd8ed1ab_0    conda-forge
   pyarrow                   3.0.0           py38h1bc9799_22_cpu    conda-forge
   r-arrow                   3.0.0             r40h9c3ff4c_1    conda-forge
   snowflake-connector-python 2.6.0            py38h411e00d_0    conda-forge
   snowflake-sqlalchemy      1.2.4              pyh9f0ad1d_0    conda-forge
   ```
4. What did you do?
   Use Pandas `to_sql`

Error:

```
Traceback (most recent call last):
  ...
    df.to_sql(
  File "/opt/conda/envs/x/lib/python3.8/site-packages/pandas/core/generic.py", line 2779, in to_sql
    sql.to_sql(
  File "/opt/conda/envs/x/lib/python3.8/site-packages/pandas/io/sql.py", line 601, in to_sql
    pandas_sql.to_sql(
  File "/opt/conda/envs/x/lib/python3.8/site-packages/pandas/io/sql.py", line 1411, in to_sql
    table.insert(chunksize, method=method)
  File "/opt/conda/envs/x/lib/python3.8/site-packages/pandas/io/sql.py", line 845, in insert
    exec_insert(conn, keys, chunk_iter)
  File "/opt/conda/envs/x/lib/python3.8/site-packages/snowflake/connector/pandas_tools.py", line 272, in pd_writer
    write_pandas(
  File "/opt/conda/envs/x/lib/python3.8/site-packages/snowflake/connector/pandas_tools.py", line 173, in write_pandas
    cursor.execute(upload_sql, _is_internal=True)
  File "/opt/conda/envs/x/lib/python3.8/site-packages/snowflake/connector/cursor.py", line 734, in execute
    data = sf_file_transfer_agent.result()
  File "/opt/conda/envs/x/lib/python3.8/site-packages/snowflake/connector/file_transfer_agent.py", line 707, in result
    Error.errorhandler_wrapper(
  File "/opt/conda/envsx/lib/python3.8/site-packages/snowflake/connector/errors.py", line 254, in errorhandler_wrapper
    handed_over = Error.hand_to_other_handler(
  File "/opt/conda/envs/x/lib/python3.8/site-packages/snowflake/connector/errors.py", line 309, in hand_to_other_handler
    cursor.errorhandler(connection, cursor, error_class, error_value)
  File "/opt/conda/envs/x/lib/python3.8/site-packages/snowflake/connector/errors.py", line 188, in default_errorhandler
    raise error_class(
snowflake.connector.errors.OperationalError: <exception str() failed>
```

I have a log file with the recommended logging configuration that I can share privately.

Note that we are in a "privatelink" environment with no other Internet access.

The problem goes away with `"use_new_put_get": False`, so it might be identical to #853.

**Comment by sfc-gh-mkeller** [comment]
Hi, @jonashaag what deployment are you using? How to find it in our UI: https://github.com/snowflakedb/snowflake-connector-python/issues/850#issuecomment-912105464 Once you share it with me I can verify that #846 solves your issue.
Please see other workarounds I have posted here: https://github.com/snowflakedb/snowflake-connector-python/issues/850#issuecomment-914484810
Note that we'll be doing a release once #846 is merged to mitigate all these issues. I apologize for the inconvenience this has caused you!

**Comment by sfc-gh-mkeller** [comment]
Let me close this, as this is technically a duplicate of #850
Please reopen if you think closing it was a mistake
